src/herder/TxSetFrame.cpp

```diff
@@ -899,12 +1778,29 @@ ApplicableTxSetFrame::checkValid(Application& app,
             return true;
         };
 
-        if (!checkFeeMap(getInclusionFeeMap(TxSetPhase::CLASSIC)))
+        for (auto const& phase : mPhases)
+        {
+            if (!checkFeeMap(phase.getInclusionFeeMap()))
+            {
+                return false;
+            }
+        }
+        if (mPhases[static_cast<size_t>(TxSetPhase::CLASSIC)].isParallel())
         {
+            CLOG_DEBUG(Herder,
+                       "Got bad txSet: classic phase can't be parallel");
             return false;
         }
-        if (!checkFeeMap(getInclusionFeeMap(TxSetPhase::SOROBAN)))
+        bool needParallelSorobanPhase = protocolVersionStartsFrom(
+            lcl.header.ledgerVersion, PARALLEL_SOROBAN_PHASE_PROTOCOL_VERSION);
+        if (mPhases[static_cast<size_t>(TxSetPhase::SOROBAN)].isParallel() !=
```

Comment:
Should we assert `mPhases` size before we index into it?

Reply:
Yeah, we should always have 2 phases by construction, but I guess an extra assertion wouldn't hurt just in case.